app/Jobs/FetchCollectionMetadataJob.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Jobs;
+
+use App\Data\Web3\Web3ContractMetadata;
+use App\Jobs\Traits\RecoversFromProviderErrors;
+use App\Models\Collection;
+use App\Models\Network;
+use App\Support\Facades\Alchemy;
+use App\Support\Queues;
+use DateTime;
+use Illuminate\Bus\Batchable;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+
+class FetchCollectionMetadataJob implements ShouldBeUnique, ShouldQueue
+{
+    use Batchable, Dispatchable, InteractsWithQueue, Queueable, RecoversFromProviderErrors;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param  array<string>  $collectionAddresses
+     */
+    public function __construct(
+        public array $collectionAddresses,
+        public Network $network,
+    ) {
+        $this->onQueue(Queues::SCHEDULED_COLLECTIONS);
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+        $metadata = Alchemy::getContractMetadataBatch($this->collectionAddresses, $this->network);
+
+        $collections = Collection::query()
+            ->whereIn('address', $metadata->pluck('contractAddress'))
+            ->select(['id', 'address'])
+            ->get();
+
+        $metadata
+            ->each(function (Web3ContractMetadata $data) use ($collections) {
+                $address = Str::lower($data->contractAddress);
+
+                $collection = $collections->first(fn ($collection) => $address === Str::lower($collection->address));
+
+                if ($collection) {
+                    Log::info('Updating collection metadata', [
+                        'collection_address' => $address,
+                        'name' => $data->collectionName,
+                        'total_supply' => $data->totalSupply,
+                        'floor_price' => $data->floorPrice,
+                        'description' => $data->description,
+                        'banner' => $data->bannerImageUrl,
+                    ]);
+
+                    $collection->name = $data->collectionName;
+                    $collection->supply = $data->totalSupply;
+                    $collection->minted_block = $data->mintedBlock;
+
+                    if ($data->description) {
```

Comment:
Sometimes Alchemy API returns response without the `openSea` property in it, that is I added these `if` checks to ensure that we are not nulling values unnecessarily:

Expected:
```JSON
  {
    "address": "0xe785e82358879f061bc3dcac6f0444462d4b5330",
    "contractMetadata": {
      "name": "World Of Women",
      "symbol": "WOW",
      "totalSupply": "10000",
      "tokenType": "ERC721",
      "contractDeployer": "0xc9b6321dc216d91e626e9baa61b06b0e4d55bdb1",
      "deployedBlockNumber": 12907782,
      "openSea": {
        "floorPrice": 0.6021,
        "collectionName": "World of Women",
        "collectionSlug": "world-of-women-nft",
        "safelistRequestStatus": "verified",
        "imageUrl": "https://i.seadn.io/gcs/files/8604de2d9aaec98dd389e3af1b1a14b6.gif?w=500&auto=format",
        "description": "World of Women is a collection of 10,000 NFTs that gives you full access to our network of artists, creators, entrepreneurs, and executives who are championing diversity and equal opportunity on the blockchain.\n\nCreated and illustrated by Yam Karkai (@ykarkai), World of Women has made prominent appearances at Christie's, The New Yorker and Billboard.\n\nJoin us to receive exclusive access to NFT drops, experiences, and much more.\n\nThe Time is WoW.",
        "externalUrl": "http://worldofwomen.art",
        "twitterUsername": "worldofwomennft",
        "discordUrl": "https://discord.gg/worldofwomen",
        "bannerImageUrl": "https://i.seadn.io/gae/GHhptRLebBOWOy8kfXpYCVqsqdes-1-6I_jbuRnGTHHW6TD63CtciH75Dotfu2u8v6EmkWt-tjhkFRVLxRUwgMfKqqy5W24AolJayeo?w=500&auto=format",
        "lastIngestedAt": "2023-09-18T00:28:26.000Z"
      }
    }
  },
```


Got:
```JSON
  {
    "address": "0xe785e82358879f061bc3dcac6f0444462d4b5330",
    "contractMetadata": {
      "name": "World Of Women",
      "symbol": "WOW",
      "totalSupply": "10000",
      "tokenType": "ERC721",
      "contractDeployer": "0xc9b6321dc216d91e626e9baa61b06b0e4d55bdb1",
      "deployedBlockNumber": 12907782
    }
  },
```